cmd/mt-whisper-importer-reader/main.go

```diff
@@ -0,0 +1,339 @@
+	if *exitOnError {
+		panic(msg)
+	} else {
+		printLock.Lock()
+		fmt.Fprintln(os.Stderr, msg)
+		printLock.Unlock()
+	}
+}
+
+func log(msg string) {
+	if *verbose {
+		printLock.Lock()
+		fmt.Println(msg)
+		printLock.Unlock()
+	}
+}
+
+func processFromChan(files chan string, wg *sync.WaitGroup) {
+	client := &http.Client{}
+
+	for file := range files {
+		fd, err := os.Open(file)
+		if err != nil {
+			throwError(fmt.Sprintf("ERROR: Failed to open whisper file '%q': %q\n", file, err))
+			continue
+		}
+		w, err := whisper.OpenWhisper(fd)
+		if err != nil {
+			throwError(fmt.Sprintf("ERROR: Failed to open whisper file '%q': %q\n", file, err))
+			continue
+		}
+
+		log(fmt.Sprintf("Processing file %q", file))
+		met, err := getMetric(w, file)
+		if err != nil {
+			throwError(fmt.Sprintf("Failed to get metric: %q", err))
+			continue
+		}
+
+		b, err := met.MarshalCompressed()
+		if err != nil {
+			throwError(fmt.Sprintf("Failed to encode metric: %q", err))
+			continue
+		}
+
+		req, err := http.NewRequest("POST", *httpEndpoint, io.Reader(b))
+		if err != nil {
+			panic(fmt.Sprintf("Cannot construct request to http endpoint %q: %q", *httpEndpoint, err))
+		}
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Content-Encoding", "gzip")
+
+		_, err = client.Do(req)
+		if err != nil {
+			throwError(fmt.Sprintf("Error sending request to http endpoint %q: %q", *httpEndpoint, err))
+			continue
+		}
+	}
+	wg.Done()
+}
+
+// generate the metric name based on the file name and given prefix
+func getMetricName(file string) string {
+	// remove all leading '/' from file name
+	for file[0] == '/' {
+		file = file[1:]
+	}
+
+	return *namePrefix + strings.Replace(strings.TrimSuffix(file, ".wsp"), "/", ".", -1)
+}
+
+// pointSorter sorts points by timestamp
+type pointSorter []whisper.Point
+
+func (a pointSorter) Len() int           { return len(a) }
+func (a pointSorter) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a pointSorter) Less(i, j int) bool { return a[i].Timestamp < a[j].Timestamp }
+
+// the whisper archives are organized like a ringbuffer. since we need to
+// insert the points into the chunks in order we first need to sort them
+func sortPoints(points pointSorter) pointSorter {
+	sort.Sort(points)
+	return points
+}
+
+func getMetric(w *whisper.Whisper, file string) (*archive.Metric, error) {
+	if len(w.Header.Archives) == 0 {
+		return nil, errors.New(fmt.Sprintf("ERROR: Whisper file contains no archives: %q", file))
+	}
+
+	archives := make([]archive.Archive, 0, len(w.Header.Archives))
+	var chunkSpan uint32
+	var rowKey string
+	name := getMetricName(file)
+
+	// md gets generated from the first archive in the whisper file
+	md := getMetricData(name, int(w.Header.Archives[0].SecondsPerPoint))
+
+	for archiveIdx, archiveInfo := range w.Header.Archives {
+		if archiveIdx > 0 {
+			rowKey = api.AggMetricKey(
+				md.Id,
+				whisper.AggregationMethod(w.Header.Metadata.AggregationMethod).String(),
```

Comment:
This is interesting. We can have values here like `average`, `last` (and `unknown` but that should never happen right?)
But MT has no average series. and #557 uses `lst` instead of `last`.